app/repeating_element/repeating_element.py

```diff
@@ -0,0 +1,72 @@
+from app.schema.answer import Answer
+import logging
+from copy import deepcopy
+logger = logging.getLogger(__name__)
+
+
+class RepeatingElementException(Exception):
+    pass
+
+
+class RepeatingElement(object):
+    ''' The repeating element manager will create a new element with new ids and q-codes using the
+        repetition number as an identifier (e.g. item.id_repetition number)
+    '''
+
+    def __init__(self, questionnaire_manager):
+        self._questionnaire_manager = questionnaire_manager
+
+    def create_element(self, current_element):
+
+        next_repetition = current_element.repetition + 1
+        new_element_id = self._rename_id(current_element.id, next_repetition)
+
+        # Check to see if we can create the element
+        if not self._questionnaire_manager.schema.item_exists(new_element_id):
+            new_element = self._clone_element(current_element, new_element_id, next_repetition)
+            self._update_state(new_element)
+        elif new_element_id in self._questionnaire_manager.archive:
+            # re-attach it, if it is in the archive
+            self._questionnaire_manager.append(self._questionnaire_manager.archive[new_element_id])
```

Comment:
We shouldn't be modifying the questionnaire manager internals outside the class. This would be better moved into the questionnaire manager itself and then exposed as a function.